.github/workflows/CI.yaml

```diff
@@ -151,15 +151,54 @@ jobs:
         with:
           python-version: 3.9
 
-      - name: Install dependencies
+      - name: Upgrade pip and install tomli
         run: |
           python -m pip install --upgrade pip
           python -m pip install tomli
-          eval "python -m pip install $(./bin/min_dependencies.py)"
-          python -m pip install -e ".[test]"
+      - name: Install minimum dependencies
+        run: eval "python -m pip install $(./bin/min_dependencies.py)"
+      - name: Install anesthetic
+        run: python -m pip install -e ".[test]"
 
       - name: Test with pytest
         run: python -m pytest --cov=anesthetic tests
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1
+
+  latest-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up latest stable Python 3
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+
+      - name: Upgrade pip and install tomli
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tomli
+      - name: Install latest dependencies
+        run: eval "python -m pip install $(./bin/latest_dependencies.py)"
+      - name: Install pytest and anesthetic
+        run: |
+          python -m pip install pytest
+          python -m pip install --no-deps -e .
+
+      - name: Test with pytest
+        run: python -m pytest
+
+      - name: Check anesthetic dependencies are up to date
+        run: |
+          latest=$(pip freeze)
+          python -m pip uninstall -y -r <(pip freeze)
+          python -m pip install -e .
+          current=$(pip freeze)
+          if [ $latest != $current ] then
+             exit 1
+          fi
```

Comment:
Make this a job rather than a step? Then we directly know whether new versions exist and break or exist and work. Otherwise we will have to look into the tests to find out the reason for the test failure...

Reply:
If you like,  but the setup will have to be repeated

Reply:
Is there maybe a way of making it separate jobs without repeating the setup? Not sufficiently fluent in CI stuff...

Reply:
Pass. I'm currently concerned with making `if [ $latest != $current ]` work properly, as I haven't done very much in bash myself

Reply:
turns out single vs double quotes is very important